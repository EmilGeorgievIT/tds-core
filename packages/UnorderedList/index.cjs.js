require('./dist/index.css')
const UnorderedList = require('./dist/index.cjs')

module.exports = UnorderedList
