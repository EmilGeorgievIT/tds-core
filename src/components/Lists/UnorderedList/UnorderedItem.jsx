import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../../safeRest'
import Text from '../../Typography/Text/Text'

import styles from './UnorderedList.modules.scss'

const UnorderedItem = ({ size, listStyle, children, ...rest }) => (
  <li {...safeRest(rest)} className={styles[`${listStyle}Item`]}>
    <Text size={size || 'base'}>
      {children}
    </Text>
  </li>
)

UnorderedItem.propTypes = {
  listStyle: PropTypes.oneOf([
    'circle',
    'checkmark',
    'x'
  ]),
  size: PropTypes.oneOf([
    'medium',
    'large'
  ]),
  /**
   * The content
   */
  children: PropTypes.node.isRequired
}

UnorderedItem.displayName = 'UnorderedList.Item'

export default UnorderedItem
