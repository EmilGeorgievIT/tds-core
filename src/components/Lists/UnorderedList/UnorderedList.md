```
<UnorderedList>
  <UnorderedList.Item>Optik Internet 25
    <UnorderedList>
      <UnorderedList.Item>Optik Internet 25</UnorderedList.Item>
      <UnorderedList.Item>Internet 25</UnorderedList.Item>
      <UnorderedList.Item>Optik High Speed Turbo</UnorderedList.Item>
      <UnorderedList.Item>High Speed Turbo 25</UnorderedList.Item>
    </UnorderedList>
  </UnorderedList.Item>
  <UnorderedList.Item>Internet 25</UnorderedList.Item>
  <UnorderedList.Item>Optik High Speed Turbo</UnorderedList.Item>
  <UnorderedList.Item>High Speed Turbo 25</UnorderedList.Item>
</UnorderedList>
```
