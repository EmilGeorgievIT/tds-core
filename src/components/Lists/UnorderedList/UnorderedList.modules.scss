@import '../../../scss/settings/colours';
@import '../../../scss/settings/typography';

.base {
  padding-left: 40px;
  position: relative;
  margin-bottom: 0.5rem;
}

.item {
  font-family: $font-icons;
}

.circle {
  composes: base;
}

.circleItem::before {
  // composes: item;
  font-family: $font-icons;

  display: block;
  position: absolute;
  content: '\002022';
  color: $color-telus-purple;
  left: -21px;
}

.checkmark {
  composes: base;
}

.checkmarkItem::before {
  // composes: item;
  font-family: $font-icons;

  content: "\f101";
  color: $color-primary;
  left: -33px;
}

.x {
  composes: base;
}

.xItem::before {
  // composes: item;
  font-family: $font-icons;

  content: "\f104";
  color: $color-cardinal;
  left: -33px;
}
