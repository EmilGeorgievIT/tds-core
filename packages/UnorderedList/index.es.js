import './dist/index.css'
import UnorderedList from './dist/index.es'

export default UnorderedList
