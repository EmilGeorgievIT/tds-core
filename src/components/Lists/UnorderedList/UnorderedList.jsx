import React from 'react'
import PropTypes from 'prop-types'
import { childrenOfType } from 'airbnb-prop-types'

import safeRest from '../../../utils/safeRest'
import UnorderedItem from './UnorderedItem/UnorderedItem'

import styles from '../List.modules.scss'

const injectSizeAndListStyle = (child, size, listStyle) => (
  React.cloneElement(child, { size, listStyle })
)

/**
 * <span class="docs--badge green">wip</span>
 */
const UnorderedList = ({ listStyle, size, children, ...rest }) => {
  return (
    <ul {...safeRest(rest)} className={styles.base}>
      {React.Children.map(children, child => injectSizeAndListStyle(child, size, listStyle))}
    </ul>
  )
}

UnorderedList.propTypes = {
  /**
   * The bullet style.
   */
  listStyle: PropTypes.oneOf([
    'circle',
    'checkmark',
    'x'
  ]),
  /**
   * The font size.
   */
  size: PropTypes.oneOf([
    'medium',
    'large'
  ]),
  /**
   * The list items. Must be at least one `UnorderedList.Item`.
   */
  children: childrenOfType(UnorderedItem).isRequired
}

UnorderedList.defaultProps = {
  listStyle: 'circle',
  size: undefined
}

UnorderedList.Item = UnorderedItem

export default UnorderedList
