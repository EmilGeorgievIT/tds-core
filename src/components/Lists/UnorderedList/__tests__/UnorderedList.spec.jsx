import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import UnorderedList from '../UnorderedList'

describe('<UnorderedList />', () => {
  const defaultChildren = '<li>One</li><li>Two</li>'
  const doShallow = (props = {}, children = defaultChildren) => (
    shallow(<UnorderedList {...props}>{children}</UnorderedList>)
  )

  it('renders', () => {
    const unorderedList = doShallow()

    expect(toJson(unorderedList)).toMatchSnapshot()
  })

  it('renders an HTML ul tag', () => {
    const unorderedList = doShallow()

    expect(unorderedList).toHaveTagName('ul')
  })

  it('can have a variant', () => {
    let unorderedList = doShallow({ variant: undefined })
    expect(unorderedList).toHaveClassName('bullet')

    unorderedList = doShallow({ variant: 'checkmark' })
    expect(unorderedList).toHaveClassName('checkmark')
  })
})
