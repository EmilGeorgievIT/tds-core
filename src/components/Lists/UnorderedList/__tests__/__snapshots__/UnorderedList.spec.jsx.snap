// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnorderedList /> renders 1`] = `
<ul
  class="base"
>
  <li
    class="circle"
  >
    <div
      class="bottomMargin-2"
    >
      Lorem ipsum
    </div>
  </li>
  <li
    class="circle"
  >
    <div
      class="bottomMargin-2"
    >
      Dolor sit amet
    </div>
  </li>
</ul>
`;
